### ISSUE TITLE
Can't Impute Using Transform Dataset Using Fitted Model

### ISSUE THREAD

**Issue by jz451** [kearnz/autoimpute#34]
For a dataset I am working on a large proportion of variables have missing values and thought that first using a simple method (mean/normal distribution) on a copy of the dataset then fitting a stochastic regression imputation would work. 

`pred = {some_large_dictionary} `
`imp = SingleImputer(strategy='stochastic',predictors=pred,copy=False).fit(df3) `
`reg_imp = imp.transform(df2)`

But when I try to transform the original dataset I get the following:

```
ValueError                                Traceback (most recent call last)
<ipython-input-179-b8b330716e07> in <module>
     12              'total_bal_ex_mort':'stochastic','total_bc_limit':'stochastic','total_il_high_credit_limit':'stochastic'}
     13 imp = SingleImputer(strategy=strat_dict,predictors=pred,copy=False).fit(df3)
---> 14 reg_imp = imp.transform(df2)

~\Anaconda3\lib\site-packages\autoimpute\utils\checks.py in wrapper(d, *args, **kwargs)
     59             err = f"Neither {d_err} nor {a_err} are of type pd.DataFrame"
     60             raise TypeError(err)
---> 61         return func(d, *args, **kwargs)
     62     return wrapper
     63 

~\Anaconda3\lib\site-packages\autoimpute\utils\checks.py in wrapper(d, *args, **kwargs)
    124 
    125         # return func if no missingness violations detected, then return wrap
--> 126         return func(d, *args, **kwargs)
    127     return wrapper
    128 

~\Anaconda3\lib\site-packages\autoimpute\utils\checks.py in wrapper(d, *args, **kwargs)
    171             err = f"All values missing in column(s) {nc}. Should be removed."
    172             raise ValueError(err)
--> 173         return func(d, *args, **kwargs)
    174     return wrapper
    175 

~\Anaconda3\lib\site-packages\autoimpute\imputations\dataframe\single_imputer.py in transform(self, X)
    259 
    260             # perform imputation given the specified imputer and value for x_
--> 261             X.loc[imp_ix, column] = imputer.impute(x_)
    262         return X
    263 

~\Anaconda3\lib\site-packages\autoimpute\imputations\series\linear_regression.py in impute(self, X)
    156         check_is_fitted(self, "statistics_")
    157         mse = self.statistics_["param"]
--> 158         preds = self.lm.predict(X)
    159 
    160         # add random draw from normal dist w/ mean squared error

~\Anaconda3\lib\site-packages\sklearn\linear_model\base.py in predict(self, X)
    219             Returns predicted values.
    220         """
--> 221         return self._decision_function(X)
    222 
    223     _preprocess_data = staticmethod(_preprocess_data)

~\Anaconda3\lib\site-packages\sklearn\linear_model\base.py in _decision_function(self, X)
    202         check_is_fitted(self, "coef_")
    203 
--> 204         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
    205         return safe_sparse_dot(X, self.coef_.T,
    206                                dense_output=True) + self.intercept_

~\Anaconda3\lib\site-packages\sklearn\utils\validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)
    540         if force_all_finite:
    541             _assert_all_finite(array,
--> 542                                allow_nan=force_all_finite == 'allow-nan')
    543 
    544     if ensure_min_samples > 0:

~\Anaconda3\lib\site-packages\sklearn\utils\validation.py in _assert_all_finite(X, allow_nan)
     54                 not allow_nan and not np.isfinite(X).all()):
     55             type_err = 'infinity' if allow_nan else 'NaN, infinity'
---> 56             raise ValueError(msg_err.format(type_err, X.dtype))
     57     # for object dtype data, we only check for NaNs (GH-13254)
     58     elif X.dtype == np.dtype('object') and not allow_nan:

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```

How do I go about solving this as the error also understandably occurs when initially trying to use fit_transform on the original dataset?

**Comment by kearnz** [comment]
this may have been fixed in version 0.11.7.

please upgrade at let me know if the issue persists.

**Comment by kearnz** [comment]
closing issue - please let me know if you're still dealing with any problems.
